packages/react-components/src/components/knowledge-graph/KnowledgeGraphPanel.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useCallback, useMemo, useState } from 'react';
+import { ElementDefinition } from 'cytoscape';
+import { Button, Container, Header, Input, SpaceBetween } from '@cloudscape-design/components';
+import { TwinMakerKGQueryDataModule } from '@iot-app-kit/source-iottwinmaker';
+import { Graph } from './graph';
+import StateManager, { useKnowledgeGraphState } from './StateManager';
+import { createKnowledgeGraphQueryClient } from './KnowledgeGraphQueries';
+import { ResponseParser } from './responseParser';
+import { getElementsDefinition } from './utils';
+
+interface KnowledgeGraphInterface {
+  kgDataSource: TwinMakerKGQueryDataModule;
+}
+const MAX_NUMBER_HOPS = 10;
+
+const KnowledgeGraphContainer: React.FC<KnowledgeGraphInterface> = ({ kgDataSource }) => {
+  const { selectedGraphNodeEntityId, setQueryResult, queryResult, clearGraphResults } = useKnowledgeGraphState();
+  const [searchTerm, setSearchTerm] = useState('');
+  //const [numOfHops, setNumOfHops] = useState(MAX_NUMBER_HOPS); // number of hops can be changed in settings which will come in the coming PRs.
+  const [elements, setElements] = useState<ElementDefinition[]>([]);
+
+  const knowledgeGraphQueryClient = useMemo(() => {
+    return createKnowledgeGraphQueryClient(kgDataSource, setQueryResult);
+  }, [kgDataSource, setQueryResult]);
+
+  const onSearchClicked = useCallback(() => {
+    if (searchTerm) {
+      knowledgeGraphQueryClient.findEntitiesByName(searchTerm);
```

Comment:
This is by design since not all queries and query results will be coming from knowledge graph component they could come from events from other components like scene viewer. This unify the rerender trigger and is based solely on change in the query results. The async/await logic is included in the createKnowledgeGraphQueryClient. However we can rethink this in the future.

Reply:
exactly this resolved by the queryResults and setQueryResults which both exists in the context. and createKnowledgeGraphQueryClient uses  setQueryResults but doesn't have to be itself part of the context.